cellfinder_napari/thread_worker.py

```diff
@@ -9,17 +10,58 @@
 )
 
 
-@thread_worker
-def run(
-    data_inputs: DataInputs,
-    detection_inputs: DetectionInputs,
-    classification_inputs: ClassificationInputs,
-    misc_inputs: MiscInputs,
-) -> list:
-    """Runs cellfinder in a separate thread, to prevent GUI blocking."""
-    return cellfinder_run(
-        **data_inputs.as_core_arguments(),
-        **detection_inputs.as_core_arguments(),
-        **classification_inputs.as_core_arguments(),
-        **misc_inputs.as_core_arguments(),
-    )
+class MyWorkerSignals(WorkerBaseSignals):
+    # Emits (label, max, value) for the progress bar
+    update_progress_bar = Signal(str, int, int)
+
+
+class Worker(WorkerBase):
+    """
+    Runs cellfinder in a separate thread, to prevent GUI blocking.
+
+    Also handles callbacks between the worker thread and main napari GUI thread
+    to update a progress bar.
+    """
+
+    def __init__(
+        self, data_inputs, detection_inputs, classification_inputs, misc_inputs
+    ):
+        super().__init__(SignalsClass=MyWorkerSignals)
+        self.data_inputs = data_inputs
+        self.detection_inputs = detection_inputs
+        self.classification_inputs = classification_inputs
+        self.misc_inputs = misc_inputs
+
+        self.npoints_detected = None
+
+    def work(self) -> list:
```

Comment:
what type of `list`?

Reply:
It's `list[Point]`, but `Point` is defined here and doesn't seem importable from Python:
https://github.com/brainglobe/cellfinder-core/blob/85127886581ef858b5607a444c80e254b4b41466/cellfinder_core/detect/filters/typedefs.pxd#L8-L11

Should I open an issue in `cellfinder-core` to expose `Point` so we can add it to the type hint here?

Reply:
I guess it's not a bad idea, I see no reason for it to be hidden in `cython`